### ISSUE TITLE
Dynamic Opengraph & Twitter images does not get generated for dynamic paths

### ISSUE THREAD

**Issue by anonrig** [vercel/next.js#48218]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 19.8.1
      npm: 9.5.1
      Yarn: 1.22.19
      pnpm: 8.1.1
    Relevant packages:
      next: 13.3.1-canary.4
      eslint-config-next: 13.3.0
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Image optimization (next/image, next/legacy/image), Metadata (metadata, generateMetadata, next/head, head.js), Middleware / Edge (API routes, runtime)

### Link to the code that reproduces this issue

https://github.com/anonrig/yagiz.co/blob/main/app/%5Bslug%5D/twitter-image.tsx

### To Reproduce

The code is available on https://github.com/anonrig/yagiz.co

### Describe the Bug

On top of the images not working on dynamic path, I'm also receiving the following error message, that may or may not be related to this issue.

```
Error: invariant: Expected pageData to be a string for app data request but received undefined. This is a bug in Next.js.
    at NextNodeServer.renderToResponseWithComponentsImpl (/var/task/node_modules/.pnpm/next@13.3.0_@opentelemetry+api@1.1.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/base-server.js:1141:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async NextNodeServer.renderPageComponent (/var/task/node_modules/.pnpm/next@13.3.0_@opentelemetry+api@1.1.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/base-server.js:1199:24)
    at async NextNodeServer.renderToResponseImpl (/var/task/node_modules/.pnpm/next@13.3.0_@opentelemetry+api@1.1.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/base-server.js:1230:32)
    at async NextNodeServer.pipeImpl (/var/task/node_modules/.pnpm/next@13.3.0_@opentelemetry+api@1.1.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/base-server.js:548:25)
    at async Object.fn (/var/task/node_modules/.pnpm/next@13.3.0_@opentelemetry+api@1.1.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/next-server.js:1030:21)
    at async Router.execute (/var/task/node_modules/.pnpm/next@13.3.0_@opentelemetry+api@1.1.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/router.js:297:32)
    at async NextNodeServer.runImpl (/var/task/node_modules/.pnpm/next@13.3.0_@opentelemetry+api@1.1.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/base-server.js:522:29)
    at async NextNodeServer.handleRequestImpl (/var/task/node_modules/.pnpm/next@13.3.0_@opentelemetry+api@1.1.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/base-server.js:456:20) {
  page: '/announcing-ada-url-parser-v2-0'
}
RequestId: 52fcf014-c6c0-4ca8-b3e7-0e8d9789fb73 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

### Expected Behavior

Twitter image:
- https://yagiz.co/performance-tips-for-c-developers/twitter-image?5bac17a4b6ff58aa

Opengraph image:
- https://yagiz.co/performance-tips-for-c-developers/opengraph-image?72bc4c3878be9510

### Which browser are you using? (if relevant)

Chrome, Safari, Arc

### How are you deploying your application? (if relevant)

Vercel

**Comment by anonrig** [comment]
By the way, locally both of these images work, but on Vercel platform, it does not work at all.

**Comment by jasongerbes** [comment]
> Opengraph image:
>
> https://yagiz.co/performance-tips-for-c-developers/opengraph-image?72bc4c3878be9510

@anonrig that image seems to load now. Have you made any changes in your repo to fix it?

**Comment by jasongerbes** [comment]
This issue appears to be resolved for me in `13.3.2-canary.1`  🎉
